x-element.js

```diff
@@ -291,31 +293,34 @@ export default class XElement extends HTMLElement {
     const attributes = new Set();
     const inputMap = new Map();
     for (const [key, property] of entries) {
-      // Attribute names are case-insensitive — lowercase to properly check for duplicates.
-      const attribute = property.attribute || XElement.__camelToKebab(key);
-      if (attribute !== attribute.toLowerCase()) {
-        throw new Error(`${path}.${key} has non-standard attribute casing "${attribute}" (use lower-cased names).`);
-      }
-      if (attributes.has(attribute)) {
-        throw new Error(`${path}.${key} causes a duplicated attribute "${attribute}".`);
-      }
-      if (PROBLEMATIC_ATTRIBUTES.has(attribute)) {
-        const { xElementPropertyName, inheritedPropertyName } = PROBLEMATIC_ATTRIBUTES.get(attribute);
-        if (xElementPropertyName) {
-          throw new Error(`Unexpected key "${path}.${key}" has attribute "${attribute}" which is related to an x-element property "${xElementPropertyName}".`);
-        } else if (inheritedPropertyName) {
-          console.warn(`Unexpected key "${path}.${key}" has attribute "${attribute}" which is related to the inherited property "${inheritedPropertyName}", behavior not guaranteed.`); // eslint-disable-line no-console
-        } else {
-          console.warn(`Unexpected key "${path}.${key}" has attribute "${attribute}" which is inherited, behavior not guaranteed.`); // eslint-disable-line no-console
+      if (!property.internal) {
```

Comment:
nit: bail early instead to cut down on the indent
```
if (property.internal) {
  return;
}
```
instead

Reply:
Hmm, I often prefer to leave the indent since it makes the code path easier to reason about — in this case using `continue` is akin to having a function with more than one `return` statement.

Also, I think in this case, that would cause a bug since the `if (property.input)` condition _can still be met_ even if the property is internal.